functional_tests/steps/release_page.py

```diff
@@ -5,216 +5,229 @@
-    else:
-        raise ValueError(f"Unsupported MeridiemIndicator: {meridiem_indicator}")
-
-
-@step('the date placeholder, "{time}", is displayed in the date input textboxes')
-def date_placeholder_is_displayed_in_release_page_date_input_fields(context: Context, time: str):
-    expect(context.page.locator("#id_release_date")).to_have_attribute("placeholder", time)
-    expect(context.page.locator("#id_next_release_date")).to_have_attribute("placeholder", time)
-
-
-@step("the user adds a release date text")
-def add_release_date_with_text(context: Context):
-    context.page.get_by_label("Or, release date text").fill("March 2025 to August 2025")
+    expected_texts = {
+        "am": "March 2025 10:00am",
+        "pm": "March 2025 5:00pm",
+    }
+    try:
+        expected = expected_texts[meridiem_indicator]
+    except KeyError as exc:
+        raise ValueError(f"Unsupported MeridiemIndicator: {meridiem_indicator}") from exc
+    expect(context.page.get_by_text(expected)).to_be_visible()
 
 
-@then("the release date text is displayed")
-def release_date_text_is_displayed(context: Context):
-    expect(context.page.get_by_text("March 2025 to August 2025")).to_be_visible()
+@step('the user sets the page status to "{page_status}"')
+def set_page_status(context: Context, page_status: str):
+    context.page.get_by_label("Status*").select_option(page_status.upper())
 
 
-@step("the user adds a next release date text")
-def add_next_release_date_with_text(context: Context):
-    context.page.get_by_label("Or, next release date text").fill("To be confirmed")
+# Combined step definition for both page status and preview mode display
+@then('the "{status_or_mode}" page is displayed')
+@then('the "{status_or_mode}" page is displayed in the preview tab')
+def display_release_page_by_status_or_mode(context: Context, status_or_mode: str):
+    match status_or_mode:
+        case "Provisional" | "Confirmed":
+            expect(context.page.get_by_text("This release is not yet published")).to_be_visible()
+        case "Cancelled":
+            expect(context.page.get_by_text("Cancelled", exact=True)).to_be_visible()
+            expect(context.page.get_by_text("Notice cancelled")).to_be_visible()
+        case "Published":
+            display_example_content_release_page_in_preview_tab(context)
+        case _:
+            raise ValueError(f"Unsupported status or mode: {status_or_mode}")
+
+
+@step("the user adds {feature} to the release calendar page")
+def add_feature(context: Context, feature: str):
+    try:
+        action = FEATURE_ACTIONS[feature]
+    except KeyError as exc:
+        raise ValueError(f"Unsupported page feature: {feature!r}") from exc
+    action(context)
+
+
+@then("the example content is displayed in the preview tab")
+def display_example_content_release_page_in_preview_tab(context: Context):
+    expect(context.page.get_by_role("heading", name="My Release")).to_be_visible()
```

Comment:
Nit: We often create an alias for `context.page` using `page` to make the code a bit less wordy, e.g.:
```python
page = context.page
expect(page.get_by_role("heading", name="My Release")).to_be_visible())
...
```
Worth doing in functions which reference `context.page` a lot